### ISSUE TITLE
dash-to-panel Gnome extension broken in 20.09

### ISSUE THREAD

**Issue by Dehumanizer77** [NixOS/nixpkgs#105001]
**Describe the bug**
In NixOS 20.09, the dash-to-panel Gnome extension stopped working and produces unexpected behaviour

**To Reproduce**
1. install gnomeExtensions.dash-to-panel package along with gnome3 desktop manager
2. log in 
3. run any application

**Expected behavior**
The application works normally.

**Screenshots**
![gnome](https://user-images.githubusercontent.com/1876451/100329566-1fe80580-2fce-11eb-912c-7a365d4d8d95.png)

**Additional context**
When an application is launched, it can't be interacted with, clicking anywhere within the application doesn't do anything, the window is in the foreground and focus is on the small X in the blue circle with the application name (see screenshot), which, if clicked, closes the application.

**Notify maintainers**
@Mounium 

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.78, NixOS, 20.09.2016.19db3e5ea27 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.9`
 - channels(root): `"nixos-20.09.2016.19db3e5ea27"`
 - channels(johny): `"nixos-20.09.2016.19db3e5ea27, unstable-21.03pre253635.2247d824fe0"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by isobit** [comment]
TL;DR this should be fixed in v39 of dash-to-panel, dash-to-panel v38 and GNOME 3.36.5 are incompatible.

Ran into the same thing I but I think I figured out the issue. Found this in my journal:
```
.gnome-shell-wr[2743]: JS ERROR: TypeError: this._syncWorkspacesFullGeometry is not a function
                                                 _newUpdateWorkspacesViews@/run/current-system/sw/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panelManager.js:563>
                                                 show@resource:///org/gnome/shell/ui/workspacesView.js:612:14
                                                 show@resource:///org/gnome/shell/ui/viewSelector.js:276:33
                                                 _animateVisible@resource:///org/gnome/shell/ui/overview.js:580:27
                                                 show@resource:///org/gnome/shell/ui/overview.js:566:14
                                                 toggle@resource:///org/gnome/shell/ui/overview.js:689:18
                                                 _initializeUI/<@resource:///org/gnome/shell/ui/main.js:220:22
```
From what I can tell digging around in the dash-to-panel issues, here's what happened:
1. GNOME 3.36.4 contained a breaking change which affected dash-to-panel v37 and below
2. dash-to-panel released a fix with v38 to work with GNOME 3.36.4 (https://github.com/home-sweet-gnome/dash-to-panel/issues/1074)
3. GNOME 3.36.5 reverted the breaking change, breaking dash-to-panel v38 (https://github.com/home-sweet-gnome/dash-to-panel/issues/1139)
4. dash-to-panel releases another fix with v39 to work with GNOME 3.36.5 (https://github.com/home-sweet-gnome/dash-to-panel/releases/tag/v39)

@Dehumanizer77 there may be a way to install a more up-to-date version of dash-to-panel (e.g. from nixos-unstable) as a workaround although I haven't looked into this myself yet.

**Comment by eduardosm** [comment]
What's keeping from updating dash-to-panel in 20.09?

**Comment by jtojnar** [comment]
Someone needs to bacport the update and make sure it works :woman_shrugging: cc @mounium (listed as maintainer) and @buckley310 who last updated it.

**Comment by buckley310** [comment]
If anyone here feels like testing the PR and posting their results over there, that's the next step in getting this fixed.

**Comment by Dehumanizer77** [comment]
@isobit I can confirm installing version 40 from nixos-unstable does fix the issue.

**Comment by buckley310** [comment]
20.09 has been superseded. Closing.
